docs/design/Network Topology Auto Discovery.md

```diff
@@ -0,0 +1,224 @@
+
+![network topology auto discovery example1](images/network-topology-aware/network-autodiscovery-nodelables-example1.png)
+
+From these node labels, We can derive a spine-leaf HyperNode topology three
+
+![network topology auto discovery example1](images/network-topology-aware/network-autodiscovery-nodelables-example1-topology.png)
+
+
+## Goals
+- **Configuration Loading**: Load the configuration containing key-value pairs for the node labels that should be recognized.
+
+- **Automatic HyperNode Creation or Adjustment**: Automatically discover network topologies based on node labels and create, delete, or update HyperNodes as necessary to ensure compliance with the desired state.
+
+## Implementation
+- **Configuration**:
+Configurations are stored in the ConfigMap named `volcano-networktopologyaware-configuration`, data structure:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: volcano-networktopologyaware-configuration
+  namespace: ${deploy_namespace}
+data:
+  node-label-based-topologies: |-
+  topologies:
+  - name: topology1
+    topo-level-keys:
+      - network-tier1  #lowest topology level
+      - network-tier2
+      - network-tier3  #highest topology level
+```
+
+- **Sync Controller**:
+The workflow of Sync controller can be summarized in the following steps:
+
+  - **Configuration Loading**
+The controller loads the configuration from a specific ConfigMap named `volcano-networktopologyaware-configuration`, which includes the key-value pairs for the node labels to be recognized.
+
+  - **Periodic Sync and HyperNode Resource Check**
+At regular intervals（default: 1 second), the controller synchronizes and checks the current state of HyperNode resources to ensure they align with the expected configuration based on the node labels and the ConfigMap.
+
+  - **Automatic HyperNode Creation or Adjustment**
+If the controller finds that the HyperNodes do not match the expected configuration (e.g., a required HyperNode is missing or misconfigured), it automatically creates, deletes, or adjusts the HyperNode resources to bring them into compliance with the desired state.
+
+## How to use
+
+1. To configure the node label keys should be recognized, define them through the  `node_label_based_topologies_override` values in the `values.yaml` file of charts. The configuration must always follow a strict hierarchy, from the lowest layer to the highest layer.
+For example:
+
+```yaml
+- name: topology1
+  topo-level-keys:
+    - network-tier1
+    - network-tier2
+    - network-tier3
+```
+This configuration instructs the controller to look for nodes with the label keys `network-tier1`，`network-tier2` and `network-tier3`
+
+**_Warning: Ensure the order of the keys is correct. Incorrect ordering of the keys will generate a topology that does not match your expectations._** 
+
+2. Assume you have k8s nodes with following labels:
```

Comment:
This example has 4 nodes, but actually created 8 nodes(node1-node8), is this right?